src/database/mod.rs

```diff
@@ -335,3 +354,55 @@ pub enum CoinType {
     Unspent,
     Spent,
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum LabelItem {
+    Address(bitcoin::Address),
+    Txid(bitcoin::Txid),
+    OutPoint(bitcoin::OutPoint),
+}
+
+impl From<bitcoin::Address> for LabelItem {
+    fn from(value: bitcoin::Address) -> Self {
+        Self::Address(value)
+    }
+}
+
+impl From<bitcoin::Txid> for LabelItem {
+    fn from(value: bitcoin::Txid) -> Self {
+        Self::Txid(value)
+    }
+}
+
+impl From<bitcoin::OutPoint> for LabelItem {
+    fn from(value: bitcoin::OutPoint) -> Self {
+        Self::OutPoint(value)
+    }
+}
+
+impl Display for LabelItem {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            LabelItem::Address(a) => write!(f, "{}", a),
+            LabelItem::Txid(a) => write!(f, "{}", a),
+            LabelItem::OutPoint(a) => write!(f, "{}", a),
+        }
+    }
+}
+
+impl FromStr for LabelItem {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<LabelItem, Self::Err> {
+        if let Ok(addr) = bitcoin::Address::from_str(s) {
+            let addr = addr.assume_checked();
+            Ok(LabelItem::Address(addr))
+        } else if let Ok(txid) = bitcoin::Txid::from_str(s) {
+            Ok(LabelItem::Txid(txid))
+        } else if let Ok(outpoint) = bitcoin::OutPoint::from_str(s) {
+            Ok(LabelItem::OutPoint(outpoint))
+        } else {
+            Err("Invalid label item")
+        }
+    }
+}
```

Comment:
Thanks for doing it, i think it makes everything nicer. However i'd really like to avoid tying the `jsonrpc` module to the `database` one.

What do you think of having a `DbLabelItem` in `src/database/sqlite/schema.rs` to be used by database routines and their interface, and a `LabelItem` in `src/commands.rs` to be used as interface for clients of the library (and the `jsonrpc` module)? This follows what already exists for other instances of this.